modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/option/SabrExtrapolationRightFunction.java

```diff
@@ -0,0 +1,598 @@
+   */
+  private boolean parameterDerivativeForwardComputed;
+  /**
+   * The derivative of the three fitting parameters with respect to the SABR parameters.
+   * <p>
+   * Those parameters are computed only when and if required.
+   */
+  private double[][] parameterDerivativeSABR = new double[4][3];
+  /**
+   * Flag indicating if the parameter derivatives to SABR parameters have been computed.
+   */
+  private boolean parameterDerivativeSABRComputed;
+  /**
+   * The Black implied volatility at the cut-off strike.
+   */
+  private double volatilityK;
+  /**
+   * The price and its derivatives of order 1 and 2 at the cut-off strike.
+   */
+  private final double[] priceK = new double[3];
+  /**
+   * Matrix decomposition. 
+   */
+  private static final SVDecompositionCommons SVD = new SVDecompositionCommons();
+  /**
+   * Black function used.
+   */
+  private static final BlackPriceFunction BLACK_FUNCTION = new BlackPriceFunction();
+  /**
+   * Value below which the time-to-expiry is considered to be 0 and the price of the fitting parameters fit a price of 0 (OTM).
+   */
+  private static final double SMALL_EXPIRY = 1.0E-6;
+  /**
+   * If the time-to-expiry is smaller than {@code SMALL_EXPIRY}, the parameter 'a' is set to be this value. 
+   */
+  private static final double SMALL_PARAMETER = -1.0E4;
+  /**
+   * Value below which the price is considered to be 0.
+   */
+  private static final double SMALL_PRICE = 1.0E-15;
+
+
+  /**
+   * Creates an instance with default volatility provider. 
+   * <p>
+   * The default volatility provider is {@link SabrHaganVolatilityFunctionProvider}.
+   * 
+   * @param forward  the forward
+   * @param sabrData  the SABR formula data
+   * @param cutOffStrike  the cut-off-strike
+   * @param timeToExpiry  the time to expiration
+   * @param mu  the mu parameter
+   * @return the instance
+   */
+  public static SabrExtrapolationRightFunction of(
+      double forward,
+      SabrFormulaData sabrData,
+      double cutOffStrike,
+      double timeToExpiry,
+      double mu) {
+    return new SabrExtrapolationRightFunction(forward, sabrData, cutOffStrike, timeToExpiry, mu,
+        SabrHaganVolatilityFunctionProvider.DEFAULT);
+  }
+
+  /**
+   * Creates an instance with volatility provider specified. 
+   * 
+   * @param forward  the forward
+   * @param sabrData  the SABR formula data
+   * @param cutOffStrike  the cut-off-strike
+   * @param timeToExpiry  the time to expiration
+   * @param mu  the mu parameter
+   * @param volatilityFunction  the volatility function
+   * @return the instance
+   */
+  public static SabrExtrapolationRightFunction of(
+      double forward, SabrFormulaData
+      sabrData,
+      double cutOffStrike,
+      double timeToExpiry,
+      double mu,
+      VolatilityFunctionProvider<SabrFormulaData> volatilityFunction) {
+    return new SabrExtrapolationRightFunction(forward, sabrData, cutOffStrike, timeToExpiry, mu, volatilityFunction);
```

Comment:
When the parameters are on multiple lines, we prefer a blank line after the open brace